README.md
# Collectio Comics e-Commerce Web & Mobile Apps

e-Commerce web + mobile app project for makeshift client Collectio Comics for ND CSE SU '20 Modern Web and App Development I.

## Tech Stack

* __Frontend__ - React through [Create React App](https://github.com/facebook/create-react-app) with [Hooks](https://reactjs.org/docs/hooks-intro.html) and [React-Router](https://reactrouter.com/web/guides/quick-start) for routing.  
* __State Management__ - [Redux](https://redux.js.org/introduction/getting-started) with [Redux-Saga](https://redux-saga.js.org/docs/introduction/BeginnerTutorial.html) (previously [Redux-Thunk](https://github.com/reduxjs/redux-thunk))  
* __Database__ - NoSQL using [Firebase](https://firebase.google.com/)  
* __Styling__ - Hybrid between [Sass/SCSS](https://sass-lang.com/documentation) and CSS in JS using [styled-components](https://styled-components.com/docs)  
* __API/s__ - Payment through [Stripe](https://stripe.com/docs/api)  
* __Deployment__ - CI/CD and hosting through [Netlify](https://www.netlify.com/)  
* __Testing__ - Unit and Integration tests through [Jest](https://jestjs.io/docs/en/getting-started) and Snapshot testing (including Redux functionality) through [Enzyme](https://enzymejs.github.io/enzyme/)  
* __Mobile__ - Hybrid (iOS + Android) mobile app wrapper through [Capacitor](https://capacitorjs.com/docs/getting-started)  

![web-screenshot](https://user-images.githubusercontent.com/50670255/88740516-63f3f800-d10b-11ea-9112-c27bbedcd4bb.png)

## Web Set Up

1. Either download the codebase or clone the repo for your own personal copy.
2. Within your own personal copy, at the root directory, do the `yarn` or `yarn install` commands to download the necessary dependencies.
3. Then, do `yarn start` to start the development environment. This will be viewable through the browser in [http://localhost:3000](http://localhost:3000).
4. You can do a `yarn build` to create a production-ready version of the app ready for deployment. This will create the `build/` folder.

## Mobile Set Up

> __Note__: You must go through the Web setup first before going through mobile as the mobile apps are wrappers for the React web app using Capacitor.

1. You must do a `yarn build` in the root directory for the production-ready version of the web app. This will be needed by the mobile setup.
2. The `capacitor.config.json` file is looking specifically at the `build/` folder for the web app. With this done and the `yarn` or `yarn install` commands done from the web setup, you now have access to the Capacitor CLI tools. Simply do a `yarn cap sync` or `yarn cap copy` in the root directory of the codebase.
3. The previous commands should have updated both the `ios/` and `android/` folders. From here, you can choose to either open the ios or android app through either `yarn cap open ios` (MacOS + XCode are required as this command will the app project codebase in XCode) or `yarn cap open android` (any OS should work and only Android Studio is required as this opens the app project codebase in Android Studio), respectively.
4. If the iOS app is created, choose the respective iOS device simulator (iPhone SE 2nd Generation should be the least GPU intensive) in XCode and click the run button. You must have a proper certificate for it to run. If you want to run it on an actual physical iOS device, you must have a proper certificate.
5. If the Android app is created, you must create an Android simulator configuration within Android Studio, then choose that (Nexus 6 API 28 should be the least GPU intensive). Click the build button and once that is successful, click the run button.

## Feature 6

### Branch
* __`buenviaje-react-performance`__ (already pulled into master but branch still exists - not deleted after merge)

### Feature: React Performance
* React Lazy + Suspense for page and shop-item preview dynamic retrieval lazy-loading
* Error Boundary for error-handling dynamic page route retrieval when lazy-loading

### How To Install & Run
Within the feature branch root directory, simply follow the [Web Set Up](#web-set-up) above. Doing it from master would be fine as this will be the latest merge, but you can also download the code from the branch and then proceed with Steps 2-4.
* Main things to take away from Steps 2-4 is that be sure to be in the root directory, and then do `yarn` or `yarn install` (for Install) before doing a `yarn start` (for Run) to view the development environment through the browser at [http://localhost:3000](http://localhost:3000).
> __Note__: You can largely ignore the mobile folders and assets as those would not impede a regular web development environment.

### Expected Functionality
1. React Lazy + Suspense  
   * When visiting routes (especially image intensive ones such as `Shop` or the individual category pages `Premium Items` and `Regular Items`), you should be able to see a `Spinner` component on the page as it loads the images of the comic books in the background before seeing the entire page fully-loaded. If not, simply refresh the [http://localhost:3000](http://localhost:3000) page from the browser.  
    *  This is also done for each individual preview of a shop item on the web app. Spamming the refresh button on the browser while on either the preview page (`Shop` on header) or the individual category pages (`Premium Items` and `Regular Items`), you should be able to see 4 `Spinner` components after the first `Spinner` component. The reason the spamming may be required is that the web app and the images retrived from Firebase are not as intensive yet so the shop items load quickly so that we do not even see the 4 `Spinner` components acting as fallbacks.
2. Error Boundary  
   * Since the pages are now being lazy-loaded (essentially an `async-await` function), there will always be a possibility of an error before the page and/or the images are loaded. Therefore, this Error Boundary feature is essentially the `try-catch` block for that displaying a better UI component if this is the case.
   * Since again, the web app is fairly small and not as prone to errors during this particular data retrieval from Firebase, I've included comments in some files that if uncommented would essentially create that error for us to show this functionality. Simply go to any of the following files: `homepage.page.jsx (src/pages/homepage/homepage.page.jsx)`, `collection.page.jsx (src/pages/collection/collection.page.jsx)`, or `collections-overview.component.jsx (src/components/collections-overview/collections-overview.component.jsx)` and uncomment the following within the top of the Component instantiation: `throw Error;`. Refresh the page and go to the route that would render the Component you just gave an error to through uncommenting the `throw Error`, and it should display an image and text telling the user that something went wrong during page or shop items retrieval.

### Files Associated
* Updated to integrate lazy-loading, suspense and error boundary
  * `src/App.js`
  * `src/pages/homepage/homepage.page.jsx`
  * `src/pages/shop/shop.page.jsx`
  * `src/pages/collection/collection.page.jsx`
  * `src/components/collection-preview/collection-preview.component.jsx`
  * `src/components/collections-overview/collections-overview.component.jsx`
* Convert spinner to non-HOC version for loading fallback component
  * `src/components/with-spinner/with-spinner.component.jsx`
  * `src/components/spinner/spinner.component.jsx`
* Create error-boundary component for error fallback component
  * `src/components/error-boundary/error-boundary.styles.jsx`
  * `src/components/error-boundary/error-boundary.component.jsx`


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [[0.6.0]] - 2020-07-28
### Added
- Testing: Enzyme Snapshot Tests
- Testing: Jest Unit + Integration Component Tests
- Testing: Redux + Redux-Saga Functionality Tests

## [[0.5.0]] - 2020-07-24
### Added
- React Lazy+Suspense for page and shop-item preview lazy-loading
- Error Boundary for page routes 404s
- Add item to cart functionality Memoization
- Optimization: Delete unused assets and minify image assets
- Optimization: PWA initialization (service worker, manifest.json, https through Netlify)

## [[0.4.0]] - 2020-07-23
### Added
- Redesign and finish mobile responsive styling through styled-components
- Redesign of icons, code comments cleanup, React Hooks adoption
- Finish iOS and android build through Capacitor, update readme

## [[0.3.0]] - 2020-07-20
### Added
- Integrate redux-saga to all async and firebase functions
- Compile all firebase functions in frontend to sagas or utils
- Add clear cart at sign out method for security
- Add auto sign in after sign up method for redirection and user ease

## [[0.2.0]] - 2020-07-08
### Added
- Implement Redux and cart functionality
- Cart session persistence (memoization), finish individual collection routing, Stripe integration and refactoring
- styled-components, fix db schema, push and implement shop-data from Firebase, spinner HOC, more routing
- Initialize CHANGELOG, use redux-thunk to group shop fetching functions, refactor for Container pattern
- Add component tree diagram, uml diagram and fix db user collection pointers

## [0.1.0] - 2020-06-26
### Added
- HOC routing, update menu items, shop images, comments
- Create and style shop preview page, ingest shop JSON data
- Finish custom functions, class component use, and template functions
- Firebase authentication and user storage through google oauth, email and password sign up/sign in

[0.2.0]: https://github.com/saejhi/collectio-site/releases/tag/v0.2.0
[0.3.0]: https://github.com/saejhi/collectio-site/compare/v0.2.0...v0.3.0
[0.4.0]: https://github.com/saejhi/collectio-site/compare/v0.3.0...v0.4.0
[0.5.0]: https://github.com/saejhi/collectio-site/compare/v0.4.0...v0.5.0
[0.6.0]: https://github.com/saejhi/collectio-site/compare/v0.5.0...v0.6.0


src/redux/shop/shop.data.js
// JS Object equivalent of shop-data.json previously used
const SHOP_DATA = {
	premium: {
		id: 1,
		categoryID: 101, // Add unique identifier to serve as pointer reference
		category: "premium",
		title: "Premium Items",
		routeName: "premium",
		items: [
			{
				id: 1,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "New Mutants #98",
				imageUrl: "https://i.ibb.co/hBRGwxH/nm-98-f.jpg",
				price: 960,
				condition: "CGC 9.6 W",
				description:
					"1st appearance of Deadpool, Gideon & Vanessa as Domino.",
				detailsUrl: "http://marvel.wikia.com/wiki/New_Mutants_Vol_1_98",
				backImageUrl: "https://i.ibb.co/MMxtpN9/nm-98-b.jpg",
			},
			{
				id: 2,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "Fantastic Four #25",
				imageUrl: "https://i.ibb.co/P5dBL9L/ff-25-f.jpg",
				price: 640,
				condition: "CBCS 7.5 OW",
				description:
					"1st Hulk vs. Thing. Classic Thing vs. Hulk cover. 2nd Silver Age Appearance of Captain America. Avengers appearance.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Fantastic_Four_Vol_1_25",
				backImageUrl: "https://i.ibb.co/BLL4SS9/ff-25-b.jpg",
			},
			{
				id: 3,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "Uncanny X-Men #266",
				imageUrl: "https://i.ibb.co/VNfCW1f/uxm-266-f.jpg",
				price: 360,
				condition: "CGC 9.6 W",
				description:
					"First full appearance of Gambit. Mystique appearance.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Uncanny_X-Men_Vol_1_266",
				backImageUrl: "https://i.ibb.co/9Hj6qLg/uxm-266-b.jpg",
			},
			{
				id: 4,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "Uncanny X-Men #141",
				imageUrl: "https://i.ibb.co/QDR1RKm/xm-141-f.jpg",
				price: 300,
				condition: "CGC 9.6 W",
				description:
					"1st app. Rachel (Phoenix II), Avalanche, Destiny, Pyro & alternate future X-Men. Senator Robert Kelly appearance.",
				detailsUrl: "http://marvel.wikia.com/wiki/X-Men_Vol_1_141",
				backImageUrl: "https://i.ibb.co/qmvBLnP/xm-141-b.jpg",
			},
			{
				id: 5,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "X-Factor #24",
				imageUrl: "https://i.ibb.co/r7LQzDc/xf-24-f.jpg",
				price: 440,
				condition: "CGC SS 9.6 W",
				description:
					"Origin & 1st appearance of Archangel. Origin of Apocalypse. Horseman app. Caliban leaves X-Factor. Signed by Walt & Louise Simonson on 10/15/11.",
				detailsUrl: "http://marvel.wikia.com/wiki/X-Factor_Vol_1_24",
				backImageUrl: "https://i.ibb.co/10rmDwv/xf-24-b.jpg",
			},
			{
				id: 6,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "X-Factor #6",
				imageUrl: "https://i.ibb.co/sFDSDPq/xf-6-f.jpg",
				price: 200,
				condition: "CGC 9.6 W",
				description:
					"1st full appearance of Apocalypse (En Sabah Nur). Alliance of Evil (Tower, Frenzy, Stinger, Timeshadow) appearance.",
				detailsUrl: "http://marvel.wikia.com/wiki/X-Factor_Vol_1_6",
				backImageUrl: "https://i.ibb.co/XL2rtSV/xf-6-b.jpg",
			},
			{
				id: 7,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "Eternals #1",
				imageUrl: "https://i.ibb.co/Tg2zMyp/e-1-f.jpg",
				price: 220,
				condition: "CGC 9.6 C/OW",
				description:
					"Origin & 1st appearance of the Eternals. 1st appearance Ikaris & Kro.",
				detailsUrl: "http://marvel.wikia.com/wiki/Eternals_Vol_1_1",
				backImageUrl: "https://i.ibb.co/zfq4bFv/e-1-b.jpg",
			},
			{
				id: 8,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "Amazing Spider-Man #253",
				imageUrl: "https://i.ibb.co/n83H9kz/asm-253-f.jpg",
				price: 180,
				condition: "CGC SS 9.6 W",
				description:
					"1st appearance of the Rose. Signed by Stan Lee on 9/3/10.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Amazing_Spider-Man_Vol_1_253",
				backImageUrl: "https://i.ibb.co/X8jmCmN/asm-253-b.jpg",
			},
			{
				id: 9,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "Amazing Spider-Man #212",
				imageUrl: "https://i.ibb.co/ygMBh1V/asm-212-f.jpg",
				price: 180,
				condition: "CGC 9.6 W",
				description: "1st appearance of Hydro-Man.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Amazing_Spider-Man_Vol_1_212",
				backImageUrl: "https://i.ibb.co/9VVNjC6/asm-212-b.jpg",
			},
			{
				id: 10,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "Amazing Spider-Man #272",
				imageUrl: "https://i.ibb.co/6PyTn33/asm-272-f.jpg",
				price: 120,
				condition: "CGC 9.6 W",
				description:
					"Origin & 1st appearance of Slyde (Jalome Beacher). Puma appearance.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Amazing_Spider-Man_Vol_1_272",
				backImageUrl: "https://i.ibb.co/rFZJvNb/asm-272-b.jpg",
			},
			{
				id: 11,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "Avengers Annual #10",
				imageUrl: "https://i.ibb.co/3dPQR1F/aa-10.jpg",
				price: 70,
				condition: "VF/NM",
				description:
					"1st appearance of Rogue. Signed by Chris Claremont.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Avengers_Annual_Vol_1_10",
				backImageUrl: "",
			},
			{
				id: 12,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "Uncanny X-Men #282",
				imageUrl: "https://i.ibb.co/vHWb4pQ/uxm-282.jpg",
				price: 30,
				condition: "VF/NM",
				description:
					"1st appearance of Bishop. Signed by Whilce Portacio.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Uncanny_X-Men_Vol_1_282",
				backImageUrl: "",
			},
			{
				id: 13,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "X-Men #4",
				imageUrl: "https://i.ibb.co/k9WBrKQ/xm-4.jpg",
				price: 10,
				condition: "VF+/NM",
				description: "1st appearance of Omega Red.",
				detailsUrl: "http://marvel.wikia.com/wiki/X-Men_Vol_2_4",
				backImageUrl: "",
			},
			{
				id: 14,
				categoryID: 101, // Point to unique identifier for reference
				category: "premium",
				name: "Avengers vs. X-Men #0",
				imageUrl: "https://i.ibb.co/VMW3StK/avx-0.jpg",
				price: 40,
				condition: "NM/M",
				description: "Stephanie Hans Phoenix Force Cover Variant.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Avengers_vs._X-Men_Vol_1_0",
				backImageUrl: "",
			},
		],
	},
	regular: {
		id: 2,
		categoryID: 102, // Add unique identifier to serve as pointer reference
		category: "regular",
		title: "Regular Items",
		routeName: "regular",
		items: [
			{
				id: 15,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "X-Men Evolutionary War Annual #12",
				imageUrl: "https://i.ibb.co/nbXgQ0h/xmewa-12.jpg",
				price: 3,
				condition: "VF+/NM-",
				description: "",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Uncanny_X-Men_Annual_Vol_1_1988",
				backImageUrl: "",
			},
			{
				id: 16,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Iron Man #9",
				imageUrl: "https://i.ibb.co/Qdx2fCh/im-9.jpg",
				price: 4,
				condition: "NM/M",
				description: "Deadpool Cover Variant.",
				detailsUrl: "http://marvel.wikia.com/wiki/Iron_Man_Vol_5_9",
				backImageUrl: "",
			},
			{
				id: 17,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Annihilators #1",
				imageUrl: "https://i.ibb.co/qMxCKCH/a-1.jpg",
				price: 5,
				condition: "VF+/NM",
				description: "",
				detailsUrl: "http://marvel.wikia.com/wiki/Annihilators_Vol_1_1",
				backImageUrl: "",
			},
			{
				id: 18,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Nova #1",
				imageUrl: "https://i.ibb.co/NFmfSyr/n-1.jpg",
				price: 5,
				condition: "NM/M",
				description: "Skottie Young Baby Cover Variant.",
				detailsUrl: "http://marvel.wikia.com/wiki/Nova_Vol_5_1",
				backImageUrl: "",
			},
			{
				id: 19,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Uncanny X-Force #1",
				imageUrl: "https://i.ibb.co/JrPS0VC/uxf-1.jpg",
				price: 5,
				condition: "NM/M",
				description: "Skottie Young Baby Cover Variant.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Uncanny_X-Force_Vol_2_1",
				backImageUrl: "",
			},
			{
				id: 20,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Secret Avengers #1",
				imageUrl: "https://i.ibb.co/1mt12Lm/sa-1.jpg",
				price: 5,
				condition: "NM/M",
				description: "Skottie Young Baby Cover Variant.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Secret_Avengers_Vol_2_1",
				backImageUrl: "",
			},
			{
				id: 21,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Secret Wars #0",
				imageUrl: "https://i.ibb.co/x6L8Fk7/sw-0.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "Free Comic Book Day (FCBD) 2015.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Free_Comic_Book_Day_Vol_2015_Secret_Wars",
				backImageUrl: "",
			},
			{
				id: 22,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "The All-New, All-Different Avengers",
				imageUrl: "https://i.ibb.co/bv3JkHF/tanada.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "Free Comic Book Day (FCBD) 2015.",
				detailsUrl:
					"http://marvel.wikia.com/wiki/Free_Comic_Book_Day_Vol_2015_Avengers",
				backImageUrl: "",
			},
			{
				id: 23,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Divergence #1",
				imageUrl: "https://i.ibb.co/8x274rF/d-1.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "Free Comic Book Day (FCBD) 2015.",
				detailsUrl: "http://dc.wikia.com/wiki/Divergence_Vol_1_1",
				backImageUrl: "",
			},
			{
				id: 24,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "The New 52 Futures End #0",
				imageUrl: "https://i.ibb.co/6gRQWC7/tn52fe-0.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "Free Comic Book Day (FCBD) 2014.",
				detailsUrl:
					"http://dc.wikia.com/wiki/The_New_52:_Futures_End_FCBD_Special_Edition",
				backImageUrl: "",
			},
			{
				id: 25,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Mister Terrific #3",
				imageUrl: "https://i.ibb.co/6YBB67s/mt-3.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "",
				detailsUrl: "http://dc.wikia.com/wiki/Mister_Terrific_Vol_1_3",
				backImageUrl: "",
			},
			{
				id: 26,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Spider-Man #33",
				imageUrl: "https://i.ibb.co/8c5wdLr/sm-33.jpg",
				price: 2,
				condition: "VF/NM",
				description: "",
				detailsUrl: "http://marvel.wikia.com/wiki/Spider-Man_Vol_1_33",
				backImageUrl: "",
			},
			{
				id: 27,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "X-Men Deluxe #39",
				imageUrl: "https://i.ibb.co/S5Xf6GT/xmd-39.jpg",
				price: 2,
				condition: "VF/NM",
				description: "",
				detailsUrl: "http://marvel.wikia.com/wiki/X-Men_Vol_2_39",
				backImageUrl: "",
			},
			{
				id: 28,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Sneak Peeks January One-Shot",
				imageUrl: "https://i.ibb.co/VmyMV9X/spjos.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "",
				detailsUrl:
					"https://comicvine.gamespot.com/january-2011-sneak-peeks-1-one-shot/4000-254779/",
				backImageUrl: "",
			},
			{
				id: 29,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Savage Dragon Legacy #1",
				imageUrl: "https://i.ibb.co/1nk5yMq/sdl-1.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "Free Comic Book Day (FCBD) 2015.",
				detailsUrl:
					"https://comicvine.gamespot.com/savage-dragon-legacy-1/4000-487500/",
				backImageUrl: "",
			},
			{
				id: 30,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Rise of the Magi #0",
				imageUrl: "https://i.ibb.co/mCQWGn6/rotm-0.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "Free Comic Book Day (FCBD) 2014.",
				detailsUrl:
					"https://comicvine.gamespot.com/rise-of-the-magi-0/4000-452072/",
				backImageUrl: "",
			},
			{
				id: 31,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Freak Force #2",
				imageUrl: "https://i.ibb.co/DLx9RM6/ff-2.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "",
				detailsUrl:
					"https://comicvine.gamespot.com/freak-force-2/4000-111768/",
				backImageUrl: "",
			},
			{
				id: 32,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Free Comic Book Day 2014 Previews",
				imageUrl: "https://i.ibb.co/c6yPMNZ/fcbd2014p.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "Free Comic Book Day (FCBD) 2014.",
				detailsUrl:
					"http://www.grahamcrackers.com/products/fcbd_2014_previewsworld_spectacular_1.htm",
				backImageUrl: "",
			},
			{
				id: 33,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Rich Johnston's The Avengefuls #1",
				imageUrl: "https://i.ibb.co/FxXJWSx/rjta-1.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "",
				detailsUrl:
					"https://comicvine.gamespot.com/rich-johnstons-the-avengefuls-1/4000-357694/",
				backImageUrl: "",
			},
			{
				id: 34,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Planet of the Apes #8",
				imageUrl: "https://i.ibb.co/RPWMwmr/pota-8.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "",
				detailsUrl:
					"https://comicvine.gamespot.com/planet-of-the-apes-8-the-devils-pawn-part-4/4000-302771/",
				backImageUrl: "",
			},
			{
				id: 35,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Irredeemable #37",
				imageUrl: "https://i.ibb.co/4dZJcqw/i-37.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "",
				detailsUrl:
					"https://comicvine.gamespot.com/irredeemable-37/4000-336590/",
				backImageUrl: "",
			},
			{
				id: 36,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Fanboys VS. Zombies #4",
				imageUrl: "https://i.ibb.co/BLDsw8X/fvz-4.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "",
				detailsUrl:
					"https://comicvine.gamespot.com/fanboys-vs-zombies-4/4000-350007/",
				backImageUrl: "",
			},
			{
				id: 37,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Starborn #2",
				imageUrl: "https://i.ibb.co/XynWL1h/s-2.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "",
				detailsUrl:
					"https://comicvine.gamespot.com/starborn-2-starborn-part-2/4000-255982/",
				backImageUrl: "",
			},
			{
				id: 38,
				categoryID: 102, // Point to unique identifier for reference
				category: "regular",
				name: "Elric: The Balance Lost #1",
				imageUrl: "https://i.ibb.co/gRnX0NY/etbl-1.jpg",
				price: 2,
				condition: "VF+/NM",
				description: "",
				detailsUrl:
					"https://comicvine.gamespot.com/elric-the-balance-lost-1/4000-278120/",
				backImageUrl: "",
			},
		],
	},
};

export default SHOP_DATA;
